Mlem/Models/Content/Community/CommunityModel.swift

```diff
@@ -0,0 +1,170 @@
+    var deleted: Bool
+    var hidden: Bool
+    var postingRestrictedToMods: Bool
+    
+    // From APICommunityView
+    var blocked: Bool?
+    var subscribed: Bool?
+    var subscriberCount: Int?
+    
+    // Dates
+    let creationDate: Date
+    let updatedDate: Date?
+    
+    // URLs
+    let communityUrl: URL
+    
+    // These values are only available via GetCommunityResponse
+    var site: APISite?
+    var moderators: [APICommunityModeratorView]?
+    var discussionLanguages: [Int]?
+    var defaultPostLanguage: Int?
+    
+    init(from response: GetCommunityResponse) {
+        self.init(from: response.communityView)
+        self.site = response.site
+        self.moderators = response.moderators
+        self.discussionLanguages = response.discussionLanguages
+        self.defaultPostLanguage = response.defaultPostLanguage
+    }
+    
+    init(from communityView: APICommunityView) {
+        self.init(from: communityView.community)
+        self.subscriberCount = communityView.counts.subscribers
+        self.subscribed = communityView.subscribed != .notSubscribed ? true : false
+        self.blocked = communityView.blocked
+    }
+    
+    init(from community: APICommunity) {
+        self.community = community
+        
+        self.communityId = community.id
+        self.instanceId = community.instanceId
+        
+        self.name = community.name
+        self.displayName = community.title
+        self.description = community.description
+        
+        self.avatar = community.iconUrl
+        self.banner = community.bannerUrl
+        
+        self.nsfw = community.nsfw
+        self.local = community.local
+        self.removed = community.removed
+        self.deleted = community.deleted
+        self.hidden = community.hidden
+        self.postingRestrictedToMods = community.postingRestrictedToMods
+        
+        self.creationDate = community.published
+        self.updatedDate = community.updated
+        
+        self.communityUrl = community.actorId
+    }
+    
+    mutating func toggleSubscribe(_ callback: @escaping (_ item: Self) -> Void = { _ in }) async throws {
+        guard let subscribed, let subscriberCount else {
+            throw CommunityError.noData
+        }
+        self.subscribed = !subscribed
+        if subscribed {
+            self.subscriberCount = subscriberCount + 1
+        } else {
+            self.subscriberCount = subscriberCount - 1
+        }
+        RunLoop.main.perform { [self] in
+            callback(self)
+        }
+        do {
+            try await apiClient.followCommunity(id: communityId, shouldFollow: !subscribed)
+        } catch {
+            hapticManager.play(haptic: .failure, priority: .high)
+            let phrase = (self.subscribed ?? false) ? "unsubscribe from" : "subscribe to"
+            errorHandler.handle(
+                .init(title: "Failed to \(phrase) community", style: .toast, underlyingError: error)
+            )
+        }
```

Comment:
Both of these cases should update the tracker again to ensure it matches the server--if it succeeds, we get the latest community info, so we might as well update, and if it fails we should revert the state fake